gui/src/app/state/spend/step.rs

```diff
@@ -175,110 +188,121 @@ impl DefineSpend {
             }
         }
     }
-    fn auto_select_coins(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
-        // Set non-input values in the same way as for user selection.
-        let mut outputs: HashMap<Address<address::NetworkUnchecked>, u64> = HashMap::new();
-        for recipient in &self.recipients {
-            outputs.insert(
-                Address::from_str(&recipient.address.value).expect("Checked before"),
-                recipient.amount().expect("Checked before"),
-            );
+    /// redraft calculates the amount left to select and auto selects coins
+    /// if the user did not select a coin manually
+    fn redraft(&mut self, daemon: Arc<dyn Daemon + Sync + Send>) {
+        if !self.form_values_are_valid() || self.recipients.is_empty() {
+            return;
         }
-        let feerate_vb = self.feerate.value.parse::<u64>().unwrap_or(0);
+
+        let destinations: Vec<(SpendOutputAddress, Amount)> = self
+            .recipients
+            .iter()
+            .map(|recipient| {
+                (
+                    SpendOutputAddress {
+                        addr: Address::from_str(&recipient.address.value)
+                            .expect("Checked before")
+                            .assume_checked(),
+                        info: None,
+                    },
+                    Amount::from_sat(recipient.amount().expect("Checked before")),
+                )
+            })
+            .collect();
+
+        let coins: Vec<CandidateCoin> = if self.is_user_coin_selection {
+            self.coins
+                .iter()
+                .filter_map(|(c, selected)| {
+                    if *selected {
+                        Some(CandidateCoin {
+                            amount: c.amount,
+                            outpoint: c.outpoint,
+                            deriv_index: c.derivation_index,
+                            is_change: c.is_change,
+                            sequence: None,
+                            must_select: *selected,
+                        })
+                    } else {
+                        None
+                    }
+                })
+                .collect()
+        } else {
+            self.coins
```

Comment:
For automated coin selection, we currently only consider confirmed coins, so I think we'll need to filter for those here to keep the same behaviour (see also https://github.com/wizardsardine/liana/issues/826).

Reply:
Ok, Thanks for the info.